service-discovery/stork-consul/src/test/java/io/quarkus/ts/stork/StorkServiceDiscoveryIT.java

```diff
@@ -0,0 +1,163 @@
+    static ConsulService consul = new ConsulService();
+
+    private static final String PREFIX = "ping-";
+    private static final String PUNG_PORT = getAvailablePort();
+    private static final String PONG_PORT = getAvailablePort();
+    private static final String PONG_REPLICA_PORT = getAvailablePort();
+
+    @QuarkusApplication(classes = PungResource.class)
+    static RestService pungService = new RestService()
+            .withProperty("quarkus.http.port", PUNG_PORT)
+            .withProperty("pung-service-port", PUNG_PORT)
+            .withProperty("pung-service-host", "localhost")
+            .withProperty("stork.pung.service-discovery", "consul")
+            .withProperty("stork.pung.service-discovery.consul-port", () -> String.valueOf(consul.getPort()))
+            .withProperty("stork.pung.service-discovery.consul-host", () -> getConsultEndpoint(consul.getConsulEndpoint()));
+
+    @QuarkusApplication(classes = PongResource.class)
+    static RestService pongService = new RestService()
+            .withProperty("quarkus.http.port", PONG_PORT)
+            .withProperty("pong-service-port", PONG_PORT)
+            .withProperty("pong-service-host", "localhost")
+            .withProperty("stork.pong.service-discovery.refresh-period", "1")
+            .withProperty("stork.pong.service-discovery", "consul")
+            .withProperty("stork.pong.service-discovery.consul-port", () -> String.valueOf(consul.getPort()))
+            .withProperty("stork.pong.service-discovery.consul-host", () -> getConsultEndpoint(consul.getConsulEndpoint()));
+
+    @QuarkusApplication(classes = PongReplicaResource.class)
+    static RestService pongReplicaService = new RestService()
+            .withProperty("quarkus.http.port", PONG_REPLICA_PORT)
+            .withProperty("pong-replica-service-port", PONG_REPLICA_PORT)
+            .withProperty("pong-replica-service-host", "localhost")
+            .withProperty("stork.pong-replica.service-discovery.refresh-period", "1")
+            .withProperty("stork.pong-replica.service-discovery", "consul")
+            .withProperty("stork.pong-replica.service-discovery.consul-port", () -> String.valueOf(consul.getPort()))
+            .withProperty("stork.pong-replica.service-discovery.consul-host",
+                    () -> getConsultEndpoint(consul.getConsulEndpoint()));
+
+    @QuarkusApplication(classes = { PingResource.class, MyBackendPungProxy.class, MyBackendPongProxy.class })
+    static RestService mainPingService = new RestService()
+            .withProperty("stork.pung.service-discovery", "consul")
+            .withProperty("stork.pung.service-discovery.consul-port", () -> String.valueOf(consul.getPort()))
+            .withProperty("stork.pung.service-discovery.consul-host", () -> getConsultEndpoint(consul.getConsulEndpoint()))
+            .withProperty("stork.pong-replica.service-discovery", "consul")
```

Comment:
I see a lot of duplication in these properties. Is there any reason they are not in "application.properties"?

Reply:
we don't want this properties in `application.properties` because in this scenario we are launching three different apps. Take a look annotation: `@QuarkusApplication(classes = { PingResource.class, MyBackendPungProxy.class, MyBackendPongProxy.class })` 

Move some property into `application.property` could have an impact in other apps and `pungService` that doesn't need `pongService` properties